Core/src/Geometry/TrackingVolume.cpp

```diff
@@ -472,164 +468,158 @@ void Acts::TrackingVolume::closeGeometry(
 
 // Returns the boundary surfaces ordered in probability to hit them based on
 boost::container::small_vector<Acts::BoundaryIntersection, 4>
-Acts::TrackingVolume::compatibleBoundaries(
-    const GeometryContext& gctx, const Vector3& position,
-    const Vector3& direction, const NavigationOptions<Surface>& options,
-    const Logger& logger) const {
+Acts::TrackingVolume::compatibleBoundaries(const GeometryContext& gctx,
+                                           const Vector3& position,
+                                           const Vector3& direction,
+                                           const NavigationOptions& options,
+                                           const Logger& logger) const {
   ACTS_VERBOSE("Finding compatibleBoundaries");
-  // Loop over boundarySurfaces and calculate the intersection
-  auto excludeObject = options.startObject;
-  boost::container::small_vector<Acts::BoundaryIntersection, 4> bIntersections;
 
-  // The Limits: current, path & overstepping
-  double nearLimit = 0;
+  boost::container::small_vector<Acts::BoundaryIntersection, 4> intersections;
+
+  // The limits for this navigation step
+  double nearLimit = options.nearLimit;
   double farLimit = options.farLimit;
 
   // Helper function to test intersection
   auto checkIntersection =
-      [&](SurfaceMultiIntersection& smIntersection,
-          const BoundarySurface* bSurface) -> BoundaryIntersection {
-    for (const auto& sIntersection : smIntersection.split()) {
-      if (!sIntersection) {
+      [&](SurfaceMultiIntersection& candidates,
+          const BoundarySurface* boundary) -> BoundaryIntersection {
+    for (const auto& intersection : candidates.split()) {
```

Comment:
Funky indentation?

Reply:
yeah that looks funny but seems like it is what clang-format wants